superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx

```diff
@@ -1451,6 +1449,23 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
                 </StyledCheckbox>
               </div>
             )}
+            {dashboardFormatOptionEnabled && (
+              <>
+                <div className="inline-container">
+                  <StyledRadioGroup
+                    onChange={onFormatChange}
+                    value={reportFormat}
+                  >
+                    <StyledRadio value="PNG">
+                      {TRANSLATIONS.SEND_AS_PNG_TEXT}
+                    </StyledRadio>
+                    <StyledRadio value="PDF">
+                      {TRANSLATIONS.SEND_AS_PDF_TEXT}
+                    </StyledRadio>
+                  </StyledRadioGroup>
+                </div>
+              </>
+            )}
```

Comment:
This is duplicating the chart specific formatting logic on 1418-1439 (=increasing code maintenance debt). I think we should consolidate these two.

Reply:
Refactored code as suggested. Thanks!